libcontainer/factory_linux.go

```diff
@@ -148,8 +148,13 @@ func RootlessCgroupfs(l *LinuxFactory) error {
 // containers that use the Intel RDT "resource control" filesystem to
 // create and manage Intel RDT resources (e.g., L3 cache, memory bandwidth).
 func IntelRdtFs(l *LinuxFactory) error {
-	l.NewIntelRdtManager = func(config *configs.Config, id string, path string) intelrdt.Manager {
-		return intelrdt.NewManager(config, id, path)
+	intelrdt.Init()
+	if !intelrdt.IsCATEnabled() && !intelrdt.IsMBAEnabled() {
```

Comment:
So, I think by requiring to call Init() explicitly you change the public API. If the only user is runc itself, this is fine. If someone else (kubernetes etc) is using libcontainer/intelrtd, this is problematic.

What you should do, I guess, is to make Init private, and let some functions that need it (such as `IsCATEnabled`) etc. call it. That way, the initialization will be performed in a transparent manner, but will only be done when needed.